code/modules/atmospherics/machinery/components/components_base.dm

```diff
@@ -90,27 +90,48 @@
 			P.build_pipeline(src)
 
 /obj/machinery/atmospherics/components/proc/nullifyPipenet(datum/pipeline/reference)
+	// Crash when we call a null pipenet here. Pretty straightforward.
 	if(!reference)
 		CRASH("nullifyPipenet(null) called by [type] on [COORD(src)]")
-	var/i = parents.Find(reference)
-	reference.other_airs -= airs[i]
-	reference.other_atmosmch -= src
+
+	// Crash if the pipeline that we are supposed to be connected to doesnt have members and machineries. 
+	// How does it even connect in the first place, right?
+
 	/**
 	 *  We explicitly qdel pipeline when this particular pipeline
 	 *  is projected to have no member and cause GC problems.
 	 *  We have to do this because components don't qdel pipelines
 	 *  while pipes must and will happily wreck and rebuild everything
 	 *	again every time they are qdeleted.
 	 */
+
 	if(!length(reference.other_atmosmch) && !length(reference.members))
 		if(QDESTROYING(reference))
-			parents[i] = null
-			CRASH("nullifyPipenet() called on qdeleting [reference] indexed on parents\[[i]\]")
+			for (var/i in 1 to parents.len)
+				if (parents[i] == reference)
+					CRASH("nullifyPipenet() called on qdeleting [reference] indexed on parents\[[i]\]")
+					parents[i] = null
```

Comment:
This should come after the loop, you don't actually need the parents[i] in here, I'm honestly not sure why it was written like that in the first place.
I'm not really sure the index of the parent's list will be helpful here, as it'll most likely change directly after this bit of code runs. you can just remove it.

Reply:
awaaay it goes.